pkg/tekton/syntax/pipeline.go

```diff
@@ -94,6 +95,11 @@ type RootOptions struct {
 	Timeout Timeout `json:"timeout,omitempty"`
 	// TODO: Not yet implemented in build-pipeline
 	Retry int8 `json:"retry,omitempty"`
+	// ContainerOptions allows for advanced configuration of containers for a single stage or the whole
+	// pipeline, adding to configuration that can be configured through the syntax already. This includes things
+	// like CPU/RAM requests/limits, secrets, ports, etc. Some of these things will end up with native syntax approaches
+	// down the road.
+	ContainerOptions *corev1.Container `json:"containerOptions,omitempty"`
```

Comment:
Since this type duplicates some things for which we provide explicit support, I think it would be good to add validation to make sure you don't try to use an option that will be ignored and overwritten with other information (e.g., name, image, command, args, env, workingDir, Stdin, TTY) (or we could try to merge those values, but it seems reasonable to ask users to use our syntax for the things it supports).